gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/WriterHelper.java

```diff
@@ -77,13 +119,22 @@ static void writePropertiesFile(File rootFolder, DocumentFileUtils documentFileU
      * @param beanClass The class type of the bean
      * @param <B> type of bean
      */
-    static <B> void writeFile(File rootFolder, DocumentFileUtils documentFileUtils, JsonTools jsonTools, Map<String, B> beans, String fileName, Class<B> beanClass) {
+    static <B extends GatewayEntity> void writeFile(File rootFolder, DocumentFileUtils documentFileUtils, JsonTools jsonTools, Map<String, B> beans, String fileName, Class<B> beanClass) {
         if (beans.isEmpty()) {
             return;
         }
 
-        File configFolder = new File(rootFolder, "config");
+        File configFolder = new File(rootFolder, CONFIG_DIRECTORY);
         documentFileUtils.createFolder(configFolder.toPath());
+
+        // remap the beans by name and run pre-write methods
+        LinkedHashMap<String, B> beansByName = new LinkedHashMap<>();
```

Comment:
any reason why we use LinkedHashMap here?

Reply:
to preserve the ordering while exporting